packages/example/src/containers/Dashboard/Dashboard.tsx

```diff
@@ -42,50 +31,25 @@ export const Dashboard = () => {
         setNetwork(networkName);
       };
 
-
-    const testFunc = async (): Promise<IPolkadotAdapter | void> => {
-        const allInjected = await web3Enable('my cool dapp');
-        let MetamaskPolkadotSnap: IPolkadotAdapter;
-
-        allInjected.forEach(extension => {
-            console.log(extension.name);
-            if(extension.name==="metamask-polkadot-snap") {
-                console.log("ddaad")
-                MetamaskPolkadotSnap = extension as IPolkadotAdapter;
-                return MetamaskPolkadotSnap;
-            } else console.log( "no polkadot adapter");
-        });
-        // const MetamaskPolkadotSnap = allInjected as unknown as IPolkadotAdapter;
-        // MetamaskPolkadotSnap.getMetamaskSnapApi
-
-        
-        // return allInjected as unknown as IPolkadotAdapter;
+    const getMetamask = (item: InjectedExtension): boolean => {
+        if(item.name==="metamask-polkadot-snap") return true;
+        else return false
     }
 
-    injectMetamaskPolkadotSnapProvider(
-        "kusama"
-        );
     useEffect(() => {
         (async () => {
-            const allInjected = await web3Enable('my cool dapp');
-            console.log(allInjected);
-            const test = await testFunc();
-            console.log(test);
-            // console.log(allInjected[0].g);
-            // allInjected["metamask-js"].getMetamaskApi().configure()
-            // const allAccounts = await web3Accounts();
-            // console.log(allAccounts);
-            // const injector = await web3FromAddress('EHkbSaapJPXsL1F83jpw6dWcN8Ar8PD1JDyKoTZVKCDF4J7');
-            // console.log(injector);
-            // const allProviders = web3ListRpcProviders('metamask-polkadot-snap');
-            // console.log(allProviders);
+            
             if(state.polkadotSnap.isInstalled) {
-                // await setConfiguration({networkName: network});
-                // await addPolkadotAsset();
-                // setPublicKey(await getPublicKey());
-                // setAddress(await getAddress());
-                // setBalance(await getBalance());
-                // setLatestBlock(await getLatestBlock());
+                const allInjected = await web3Enable('my cool dapp');
```

Comment:
I don't think this is ok, you should call this here: https://github.com/NodeFactoryIo/metamask-snap-polkadot/blob/develop/packages/example/src/services/metamask.ts#L25

You should add this util method:
```javascript

export async function getInjectedMetamaskExtension(): Promise<InjectedMetamaskExtension|null> {
   return(await web3EnablePromise).find(..);
}

```
you should use this method in `isPolkadotSnapInstalled`.
In the end you should have following flow:
1. check if metamask - fail with error screen abount no metamask
2. check if polkadot snap installed
   2.1. if installed use await getInjectedMetamaskExtension to invoke api and fetch data
   2.2 if not install call `installPolkadotSnap` and proceed as usual